### ISSUE TITLE
fields named `newValue` can cause compilation failures

### ISSUE THREAD

**Issue by ldriscoll** [immutables/immutables#416]
we have an immutable template that looks like below, but it fails because the generated code for the `withNewValue` fails to compile:

template:
```java
@Immutable
public interface AbstractUpdater {
  String getNewValue();
  String getOldValue();

  @Lazy
  default boolean isDifferent() {
    return !Objects.equals(getNewValue(), getOldValue());
  }
}
```

`withNewValue` generates bad code, because there are two `String newValue`s:
```java
  public final ClientTiersDiffSection withNewValue(String newValue) {
    if (this.newValue.equals(newValue)) return this;
    String newValue = Objects.requireNonNull(newValue, "newValue");
    return new ClientTiersDiffSection(this.clientTierDiffType, newValue, this.oldValue);
  }
```

**Comment by elucash** [comment]
Yep, I've imagined this might occur, we'll just use 'value' as a parameter name as all field are accessed using `this.` in this context anyway.

**Comment by ldriscoll** [comment]
Seems like a solution.

**Comment by elucash** [comment]
released in 2.2.11 Thank you!
